### PR TITLE
codeintel: Register SCIP oobmigration

### LINKED ISSUE #45087: RFC 740: Write migration to translate LSIF -> SCIP
Create an out-of-band migration to move existing data into the new database schema defined in the [RFC](https://docs.google.com/document/d/1PuTYmHYLElYKIEW0sSAAUbXf7XtnsMdEq3EshD-O7do/edit).

### PR DESCRIPTION
**Pull request by efritz**
This PR registers the migration introduced in the linked PR.

Fixes #45087.

- [x] Must be coordinated with https://github.com/sourcegraph/src-cli/pull/897.

## Test plan

[main-dry-run](https://buildkite.com/sourcegraph/sourcegraph/builds/194439)

### CODE REVIEW

**Comment by efritz** [comment]
Related: https://github.com/sourcegraph/deploy-sourcegraph-cloud/pull/17560

**Review by Numbers88s — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,7 +13,17 @@ All notable changes to Sourcegraph are documented in this file.
 
 <!-- START CHANGELOG -->
 
-## Unreleased
+## Unreleased (4.5)
```
Comment (efritz) [inline CHANGELOG.md:16]: - [ ] Merge after 4.4 is released, update changelog.

File: doc/admin/updates/docker_compose.md [hunk 2]
```diff
@@ -16,7 +16,11 @@ This page lists the changes that are relevant for [upgrading Sourcegraph on Dock
 
 <!-- GENERATE UPGRADE GUIDE ON RELEASE (release tooling uses this to add entries) -->
 
-## Unreleased
+## Unreleased (4.5)
```
Comment (efritz) [inline doc/admin/updates/docker_compose.md:19]: - [ ] Merge after 4.4 is released, update upgrade notes.

File: doc/admin/updates/kubernetes.md [hunk 3]
```diff
@@ -20,7 +20,11 @@
 
 <!-- GENERATE UPGRADE GUIDE ON RELEASE (release tooling uses this to add entries) -->
 
-## Unreleased
+## Unreleased (4.5)
```
Comment (efritz) [inline doc/admin/updates/kubernetes.md:23]: - [ ] Merge after 4.4 is released, update upgrade notes.

File: doc/admin/updates/pure_docker.md [hunk 4]
```diff
@@ -8,7 +8,11 @@ Each section comprehensively describes the changes needed in Docker images, envi
 
 <!-- GENERATE UPGRADE GUIDE ON RELEASE (release tooling uses this to add entries) -->
 
-## Unreleased
+## Unreleased (4.5)
```
Comment (efritz) [inline doc/admin/updates/pure_docker.md:11]: - [ ] Merge after 4.4 is released, update upgrade notes.

File: doc/admin/updates/server.md [hunk 5]
```diff
@@ -17,7 +17,11 @@ This document describes the exact changes needed to update a single-node Sourceg
 
 <!-- GENERATE UPGRADE GUIDE ON RELEASE (release tooling uses this to add entries) -->
 
-## Unreleased
+## Unreleased (4.5)
```
Comment (efritz) [inline doc/admin/updates/server.md:20]: - [ ] Merge after 4.4 is released, update upgrade notes.

File: dev/codeintel-qa/internal/indexes.go [hunk 6]
```diff
@@ -7,22 +7,28 @@ import (
 	"github.com/grafana/regexp"
 )
 
-var indexFilenamePattern = regexp.MustCompile(`^([^.]+)\.([^.]+)\.\d+\.([0-9A-Fa-f]{40})\.dump$`)
+var indexFilenamePattern = regexp.MustCompile(`^([^.]+)\.([^.]+)\.\d+\.([0-9A-Fa-f]{40})\.(scip|dump)$`)
```
Comment (Strum355) [inline dev/codeintel-qa/internal/indexes.go:10]: werent the outputs usually dump.lsif, surprised this worked before?

File: CHANGELOG.md [hunk 7]
```diff
@@ -17,7 +17,7 @@ All notable changes to Sourcegraph are documented in this file.
 
 ### Added
 
--
+- Added out-of-band migration that will migrate all existing data from LSIF to SCIP (see additional [migration documentation](https://docs.sourcegraph.com/admin/how-to/lsif_scip_migration)). [#45106](https://github.com/sourcegraph/sourcegraph/pull/45106)
```
Comment (Strum355) [inline CHANGELOG.md:20]: should we add the big warning here too? People mightnt click these links

**Review by Strum355 — APPROVED** [review]

File: CHANGELOG.md (hunk 7 above)
Comment (efritz) [reply CHANGELOG.md:21]: cc @coury-clark 

Thoughts on how much attention to bring to changelog notices like these?

Comment (coury-clark) [reply CHANGELOG.md:21]: If this is something that every admin needs to read, I'd suggest highlighting this with something like `Important:`, and adding it to the release post https://github.com/sourcegraph/about/pull/5951

Comment (efritz) [reply CHANGELOG.md:21]: I can write more in the 4.5 release post.

File: dev/codeintel-qa/internal/indexes.go (hunk 6 above)
Comment (efritz) [reply dev/codeintel-qa/internal/indexes.go:10]: They were _only_ `.dump` before. The file names are hard-coded by the download/clone-and-index scripts.

**Pull request merged**